### PR TITLE
Add name for readonly records in config schema generation

### PR DESCRIPTION
**Pull request by Dilhasha**
## Purpose

Fix failure when Config Generation fails to add the name of Readonly Records to the Config Schema.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/41906

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/configschema/TypeConverter.java [hunk 1]
```diff
@@ -181,14 +181,18 @@ private JsonObject generateRecordType(BRecordType effectiveType, BIntersectionTy
         if (!requiredFields.isEmpty()) {
             typeNode.add("required", requiredFields);
         }
-        // Get record type and set the type name as a property
-        for (BType bType : intersectionType.getConstituentTypes()) {
-            // Does not consider anonymous records
-            if (bType.tag == TypeTags.TYPEREFDESC) {
-                typeNode.addProperty("name", bType.toString().trim());
+        // The tsymbol name is empty for anon intersection type created due to inferred readonly
+        if (!intersectionType.tsymbol.name.value.isEmpty()) {
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/configschema/TypeConverter.java:185]: Can probably do

```suggestion
        if (intersectionType.tsymbol.name != Names.EMPTY) {
```

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/configschema/TypeConverter.java [hunk 2]
```diff
@@ -181,14 +181,18 @@ private JsonObject generateRecordType(BRecordType effectiveType, BIntersectionTy
...
+            typeNode.addProperty("name", intersectionType.tsymbol.toString().trim());
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/configschema/TypeConverter.java:186]: Shouldn't this be tsymbol.name.value?

Comment (Dilhasha) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/configschema/TypeConverter.java:186]: `tsymbol.name.value` doesn't return the fully qualified name.

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/configschema/TypeConverter.java [hunk 3]
```diff
@@ -181,14 +182,19 @@ private JsonObject generateRecordType(BRecordType effectiveType, BIntersectionTy
         if (!requiredFields.isEmpty()) {
             typeNode.add("required", requiredFields);
         }
-        // Get record type and set the type name as a property
-        for (BType bType : intersectionType.getConstituentTypes()) {
-            // Does not consider anonymous records
-            if (bType.tag == TypeTags.TYPEREFDESC) {
-                typeNode.addProperty("name", bType.toString().trim());
+        // The tsymbol name is empty for anon intersection type created due to inferred readonly
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/configschema/TypeConverter.java:185]: Or implicitly?

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/configschema/TypeConverter.java [hunk 4]
```diff
@@ -181,14 +182,19 @@ private JsonObject generateRecordType(BRecordType effectiveType, BIntersectionTy
...
+        if (intersectionType.tsymbol.name != Names.EMPTY) {
+            typeNode.addProperty("name", intersectionType.tsymbol.toString().trim());
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/configschema/TypeConverter.java:187]: Can extract `intersectionType.tsymbol` out to a variable.

**Review by MaryamZi — COMMENTED** [review]
Rest LGTM.

**Review by azinneera — APPROVED** [review]

**Pull request merged**